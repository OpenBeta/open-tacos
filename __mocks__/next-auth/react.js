export function useSession () {
  return {
    status: 'authenticated',
    data: {}
  }
}
