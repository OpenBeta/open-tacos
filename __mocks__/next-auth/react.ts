export function useSession (): any {
  return {
    status: 'authenticated',
    data: {}
  }
}
