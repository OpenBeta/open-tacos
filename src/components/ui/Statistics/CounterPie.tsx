import { ResponsiveContainer, PieChart, Pie, Cell } from 'recharts'

export interface CounterPieProps {
  total: number
  forYou: number
}

const CounterPie = ({ total, forYou }: CounterPieProps): JSX.Element => {
  const data = [
    { name: 'total', value: total },
    { name: 'forYou', value: forYou }
  ]
  return (
    <ResponsiveContainer>
      <PieChart>
        <Pie
          data={data}
          cx='50%'
          cy='50%'
          innerRadius='60%'
          outerRadius='80%'
          startAngle={90}
          endAngle={450}
          fill='#000000'
          dataKey='value'
          paddingAngle={5}
          isAnimationActive={false}
        >
          <Cell key='total' fill='#D1D5DB' />
        </Pie>
        <g>
          <text x='50%' y='50%' fontSize='1.25rem' textAnchor='middle' alignmentBaseline='middle' fill='#000000'>
            {forYou}
          </text>
        </g>
      </PieChart>
    </ResponsiveContainer>
  )
}

// const renderCustomizedLabel = ({
//   cx,
//   cy,
//   midAngle,
//   innerRadius,
//   outerRadius,
//   percent,
//   index
// }: any) => {
//   const radius = innerRadius + (outerRadius - innerRadius) * 0.5
//   // const x = cx + radius * Math.cos(-midAngle * RADIAN)
//   // const y = cy + radius * Math.sin(-midAngle * RADIAN)

//   if (index === 0) return null
//   console.log('label', index)
//   return (
//     <text
//       x={cx}
//       y={cy}
//       // fill='white'
//       // textAnchor={x > cx ? 'start' : 'end'}
//       dominantBaseline='central'
//     >
//       {`${(percent * 100).toFixed(0)}%`}
//     </text>
//   )
// }

export default CounterPie
