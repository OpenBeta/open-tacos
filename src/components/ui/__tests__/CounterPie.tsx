import React from 'react'
import ''
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import CounterPie from '../Statistics/CounterPie'

const total = 18
const forYou = 14

// not having width and height props was breaking the component rendering in test
// solution: https://github.com/recharts/recharts/issues/727#issuecomment-758556820
test('that component will not render without width and height', () => {
  const { container } = render(<CounterPie total={total} forYou={forYou} />)
  const sectors = container.getElementsByClassName('recharts-pie-sector')
  expect(sectors.length).toBe(0)
})

test('that there are 2 parts to the pie chart', async () => {
  jest.spyOn(HTMLElement.prototype, 'clientHeight', 'get').mockReturnValue(96)
  jest.spyOn(HTMLElement.prototype, 'clientWidth', 'get').mockReturnValue(96)
  // there should only ever be 2 sectors because of data shape below
  // const data = [
  //     { name: 'total', value: total - forYou },
  //     { name: 'forYou', value: forYou }
  //   ]
  const { container } = render(<CounterPie total={total} forYou={forYou} />)
  const sectors = container.getElementsByClassName('recharts-pie-sector')
  expect(sectors.length).toBe(2)
})

it('renders correctly', () => {
  jest.spyOn(HTMLElement.prototype, 'clientHeight', 'get').mockReturnValue(96)
  jest.spyOn(HTMLElement.prototype, 'clientWidth', 'get').mockReturnValue(96)
  const tree = renderer
    .create(<div style={{ width: '100px', height: '100px' }}><CounterPie total={total} forYou={forYou} /></div>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
