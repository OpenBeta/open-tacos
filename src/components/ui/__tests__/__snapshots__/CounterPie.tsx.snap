// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div
    className="recharts-responsive-container"
    style={
      {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />
</div>
`;
